### DISCUSSION TITLE
Realistic speedups to expect with XPU

### DISCUSSION THREAD

**Discussion by boberfly** [dreamworksanimation/openmoonray#50] [Q&A] [answered]
Hi all,

I am just wondering, due to XPU only tackling occlusion rays accelerated co-processing, does this only account for pure ambient-occlusion and shadow rays? Breaking this down for a regular beauty render, this would only accelerate shadow rays right? In production, what kind of speedups does this gain realistically?

I saw an example render from one of the Moonray videos that shadow rays only make up a tiny fraction of the rendering time, so is this kind of pointless and just a proof of concept until other rays can be supported or am I missing something? Maybe it indirectly speeds up other rays or render passes?

Thank you for any insights on this, it's a curiosity of mine...

**Comment by kjeffery** [comment, marked as answer]
The speedup will vary depending on multiple factors: the scene, the number of shadow rays, and the hardware. Embree (BVH) intersections, including occlusion rays, are often the most time consuming aspect of a rendering, so there is potential for a lot of speedup. I don't have any numbers in front of me.

**Reply by boberfly** [reply]
Thanks for the answer @kjeffery that makes sense to me when using Arnold with increasing the low light threshold to reduce shadow rays and gain fairly substantial speedups so I was thinking this could be the case with Moonray as well. 

That's a clever way now that I think about it, as you use the smaller amount of VRAM purely for geometry+BVH and not need to load the shading+texturing stack...

**Reply by kjeffery** [reply]
You are onto us! You're right: we do it this way to avoid loading the entire scene into the video card memory (which we can’t do for the scenes produced at DreamWorks Animation without going to out-of-core rendering). Plus, with a lot of diligent work, we were able to get matching renders with and without GPU support.
